lib/Dialect/TTIR/IR/TTIROps.cpp

```diff
@@ -262,6 +262,49 @@ ::mlir::LogicalResult mlir::tt::ttir::ReshapeOp::verify() {
   return success();
 }
 
+::mlir::LogicalResult mlir::tt::ttir::SqueezeOp::verify() {
+  ::mlir::RankedTensorType inputType = getInput().getType();
+  ::mlir::RankedTensorType outputType = getOutput().getType();
+  int32_t dim = getDim();
+
+  // Check that the dimension `dim` is valid.
+  if (dim < 0 || dim >= inputType.getRank()) {
```

Comment:
Can we also handle negative dims? E.g. -1. 

From an FFE perspective, it's common to use negative dims to handle higher-rank tensors. Mostly because we know the structure of the last dims, but higher ranks can have different meanings from model to model (e.g. batch, input source, etc.).

_Note: This isn't a deal braker, before lowering we can covert those to positive dims. However, seems like we would have more flexibility if we can convert those here E.g.` dim =+ ranks_size`_

```

Reply:
Makes sense. #550 added item to track.

Reply:
Great! I added it to Llama 3B milestone as well. I don't have confirmation if there are cases on Llama, but will now for sure next week